ComponentKit/Core/Scope/CKComponentControllerAppearanceEvents.h

```diff
@@ -0,0 +1,17 @@
+//
+//  CKComponentControllerAppearanceEvents.h
+//  ComponentKit
+//
+//  Created by Oliver Rickard on 3/27/17.
+//
+//
+
+#import <Foundation/Foundation.h>
+
+#import <ComponentKit/CKComponentScopeRoot.h>
+
+BOOL CKComponentControllerAppearanceEventPredicate(id<CKScopedComponentController> controller);
+BOOL CKComponentControllerDisappearanceEventPredicate(id<CKScopedComponentController> controller);
+
+void CKComponentControllerAnnounceAppearance(CKComponentScopeRoot *scopeRoot);
+void CKComponentControllerAnnounceDisappearance(CKComponentScopeRoot *scopeRoot);
```

Comment:
I find this method a little confusing because you are providing a scope root. Could we name it something like:

`CKScopeRootAnnounceApperanceToControllers(CKComponentScopeRoot *scopeRoot)`